File: front/src/Entities/Coupon.ts
import User from "./User"

class Coupon {
    id: number
    store: string
    code: string
    creator: User
    owner: User | null

    constructor( 
        id: number,
        store: string,
        code: string,
        creator: User,
        owner: User | null
    ){
        this.id = id
        this.store = store
        this.code = code
        this.creator = creator
        this.owner = owner
    }
}

export default Coupon

File: Back/src/main/java/com/playful/chat/model/Coupon.java
package com.playful.chat.model;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
public class Coupon {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String code;

    @ManyToOne
    @JoinColumn(name = "creator_id")
    private UserModel creator;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private UserModel owner;

    @Column(nullable = false)
    private LocalDateTime dateCreated;

    @Column(nullable = false)
    private String store;

}
